ceph/ceph_admin/helper.py

```diff
@@ -2,10 +2,183 @@
+
+        specs:
+          - service_type: host
+            address: true
+            labels: apply-all-labels
+            nodes:
+                - node2
+                - node3
+          - service_type: mon
+            placement:
+              nodes:
+                - node2
+                - node3
+          - service_type: mgr
+            placement:
+                count: 2
+          - service_type: alertmanager
+            placement:
+                count: 1
+          - service_type: crash
+            placement:
+                host_pattern: '*'
+          - service_type: grafana
+            placement:
+                count: 1
+          - service_type: node-exporter
+            placement:
+                host_pattern: '*'
+          - service_type: prometheus
+            placement:
+                count: 1
+        """
+        self.cluster = cluster
+        self.specs = specs
+        self.template_path = dirname(__file__) + "/jinja_templates/"
+
+    def _get_template(self, service_type):
+        """
+        Return Jinja template based on the service_type
+        Args:
+            service_type: service name (ex., "host")
+        Returns:
+            template
+        """
+        path = self.template_path + f"{service_type}.jinja"
+        with open(path) as fd:
+            template = fd.read()
+        return Template(template)
+
+    def render_hosts(self, spec):
+        """
+        prepare hosts spec content based on host config
+        args:
+            spec: hosts specification
+
+        spec:
+          - service_type: host
+            address: true
+            labels: apply-all-labels
+            nodes:
+                - node2
+                - node3
+        Returns:
+            hosts_spec
+        """
+        template = self._get_template("host")
+        hosts = []
+        address = spec.get("address")
+        labels = spec.get("labels")
+        for node_name in spec["nodes"]:
+            host = dict()
+            node = get_node_by_id(self.cluster, node_name)
+            host["hostname"] = node.shortname
+            if address:
+                host["address"] = node.ip_address
+            if labels:
+                host["labels"] = node.role.role_list
+            hosts.append(host)
+
+        return template.render(hosts=hosts)
+
+    def render_common_services(self, spec):
+        """
+        prepare spec content for common services
+         which is mentioned in COMMON_SERVICES
+
+        Args:
+            spec: common service spec config
+
+        spec:
+          - service_type: mon
+            placement:
+              nodes:
+                - node2
+                - node3
+        Returns:
+        """
+        raise NotImplementedError
+
+    def render_rgw_spec(self, spec):
+        raise NotImplementedError
+
+    def render_mds_spec(self, spec):
+        raise NotImplementedError
+
+    def render_nfs_spec(self, spec):
+        raise NotImplementedError
```

Comment:
These methods can be moved to their respective child objects.

Reply:
Removed these methods for now.